src/ssl.c

```diff
@@ -20151,9 +20151,41 @@ WOLFSSL_CTX* wolfSSL_set_SSL_CTX(WOLFSSL* ssl, WOLFSSL_CTX* ctx)
     ssl->ctx = ctx;
 
 #ifndef NO_CERTS
+#ifdef WOLFSSL_COPY_CERT
+    /* If WOLFSSL_COPY_CERT defined, always make new copy of cert */
+    if (ctx->certificate != NULL) {
+        if (ssl->buffers.certificate != NULL) {
+            FreeDer(&ssl->buffers.certificate);
```

Comment:
This gets tricky.
Did the user set the certificate against the SSL explicitly?
If so, I would expect that the certificate wouldn't change.

Reply:
I tried to implement as you said, but it seems that is not compatible with expected behavior from openssl. They always set the new cert and free the old if it exists. I have changed the code back to reflect that behavior.